tripy/tripy/common/array.py

```diff
@@ -169,7 +170,7 @@ def __init__(
             self.dtype = dtype
             self.shape = shape
         else:
-            if isinstance(data, (List, bool, int, float, tuple)):
+            if not self.has_memref:
```

Comment:
Is there any way we can simplify all the branching here? Maybe don't create an `Array` at all if we're using a Python list?

Reply:
Yes I've explored that option too, I was stuck at the return type of `Tensor.eval()`, which is always an `Array`.  I will continue on this approach today.